delta-app/src/main/java/io/cdap/delta/store/DefaultMacroEvaluator.java

```diff
@@ -43,11 +43,10 @@ public DefaultMacroEvaluator(Map<String, String> runtimeArgs, SecureStore secure
   @Override
   @Nullable
   public String lookup(String property) {
-    String val = runtimeArgs.get(property);
-    if (val == null) {
-      throw new InvalidMacroException(String.format("Argument '%s' is not defined.", property));
-    }
-    return val;
+    String msg = String.format("Macro cannot be resolved for property '%s' as macro resolution is not supported " +
+                                 "for replication pipelines. Only secure store macro functions are supported.",
+                               property);
+    throw new RuntimeException(msg);
```

Comment:
better to throw an InvalidMacroException. Should avoid throwing Exception or RuntimeException in most circumstances.